### PR TITLE
VAULT-22483: Audit `filter` docs

### PR DESCRIPTION
**Pull request by peteski22**
Documentation for the audit `filter` feature in 1.16.

**Concepts**:

* `/vault/docs/concepts/filtering`
* `/vault/docs/concepts/filtering/audit`

**API**:
 
* `/vault/api-docs/system/audit#options`

**CLI**:

* `/vault/docs/commands/audit/enable#usage`

**Common configuration options**:

* `/vault/docs/audit#common-configuration-options`

**Metrics**: 

* `/vault/docs/internals/telemetry/metrics/audit`
* `/vault/docs/internals/telemetry/metrics/all#vault-audit-sink-success`

### CODE REVIEW

File: website/content/api-docs/system/audit.mdx
Comment (peteski22) [inline website/content/api-docs/system/audit.mdx:1]: Format fixing changes.

File: website/content/docs/audit/file.mdx
Comment (peteski22) [inline website/content/docs/audit/file.mdx:1]: Format fixing changes.

File: website/content/docs/audit/socket.mdx
Comment (peteski22) [inline website/content/docs/audit/socket.mdx:1]: Format fixing changes.

File: website/content/docs/commands/events.mdx
Comment (peteski22) [inline website/content/docs/commands/events.mdx:1]: Update the events page to refer to our own now instead of Boundary

File: website/content/docs/concepts/events.mdx
Comment (peteski22) [inline website/content/docs/concepts/events.mdx:1]: Format fixing changes.

File: website/content/partials/telemetry-metrics/vault/audit/log_request.mdx
Comment (peteski22) [inline website/content/partials/telemetry-metrics/vault/audit/log_request.mdx:1]: Format fixing changes.

File: website/content/partials/telemetry-metrics/vault/audit/log_request_failure.mdx
Comment (peteski22) [inline website/content/partials/telemetry-metrics/vault/audit/log_request_failure.mdx:1]: Format fixing changes.

File: website/content/partials/telemetry-metrics/vault/audit/log_response.mdx
Comment (peteski22) [inline website/content/partials/telemetry-metrics/vault/audit/log_response.mdx:1]: Format fixing changes.

File: website/content/partials/telemetry-metrics/vault/audit/log_response_failure.mdx
Comment (peteski22) [inline website/content/partials/telemetry-metrics/vault/audit/log_response_failure.mdx:1]: Format fixing changes.

File: website/content/docs/concepts/filtering/index.mdx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+Each filter expression has matching operators composed with selectors and
+values.
+
+## Create expressions
+
+A single expression is a matching operator with a selector and value. They are
+written in plain text format, and Boolean logic and parenthesization are
+supported. In general whitespace is ignored, except within literal strings.
+
+### Matching operators
```
Comment (mladlow) [inline website/content/docs/concepts/filtering/index.mdx:24]: Would it be possible to add some examples throughout, or maybe a big example at the end that puts all these things together?

File: website/content/docs/concepts/filtering/audit.mdx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+[filter syntax]: /vault/docs/concepts/filtering
+
+# Filtering audit entries
+
+This page describes how to use filters with Vault audit devices to fine tune which
+audit entries should be written to an audit log.
+
+Starting in Vault 1.16.0, audit devices can be enabled with a `filter` option which
+is used to evaluate audit entries and decide if they should be written to an audit log.
+Please ensure you are familiar with the [filtering concept](/vault/docs/concepts/filtering)
```
Comment (mladlow) [inline website/content/docs/concepts/filtering/audit.mdx:17]: This might still be forthcoming but I'd also love to get some examples here to help make everything more concrete.

File: website/content/docs/concepts/filtering/index.mdx (hunk 1 above)
Comment (peteski22) [reply website/content/docs/concepts/filtering/index.mdx:24]: This page has been taken from Boundary in order for Vault to not have to reference their page in our docs.

https://developer.hashicorp.com/boundary/docs/concepts/filtering

File: website/content/docs/concepts/events.mdx
Comment (VioletHynes) [inline website/content/docs/concepts/events.mdx:1]: Was this file meaningfully modified or is it all auto-format? I'm a little confused by the diff and thought it best to ask. It doesn't seem like it, but I might be missing something in the diff.

File: website/content/docs/concepts/filtering/audit.mdx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+Starting in Vault 1.16.0, audit devices can be enabled with a `filter` option, which
+is used to evaluate audit entries and decide if they should be written to a particular
+audit log. Please ensure you are familiar with the [filtering concept](/vault/docs/concepts/filtering) before
+attempting to configure an audit device with a filter.
+
+<Note title="Advanced audit feature">
+  The use of filtering in Vault's audit system should be considered an advanced feature.
+  Whilst it has been designed to be simple and flexible, exclusively enabling filtered
+  devices could result in some requests and responses that are not audited.
```
Comment (VioletHynes) [inline website/content/docs/concepts/filtering/audit.mdx:23]:
```suggestion
  devices without a fallback configured could result in some requests and responses that are not audited.
```

File: website/content/docs/concepts/filtering/audit.mdx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+[filter syntax]: /vault/docs/concepts/filtering
+
+# Filtering audit entries
+
+This page describes how to use filters with Vault audit devices to fine tune which
+audit entries should be written to an audit log.
+
+Starting in Vault 1.16.0, audit devices can be enabled with a `filter` option, which
+is used to evaluate audit entries and decide if they should be written to a particular
```
Comment (VioletHynes) [inline website/content/docs/concepts/filtering/audit.mdx:16]:
```suggestion
is used to evaluate audit entries and determine if they should be written to a particular
```

File: website/content/docs/concepts/filtering/audit.mdx [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+audit log. Please ensure you are familiar with the [filtering concept](/vault/docs/concepts/filtering) before
+attempting to configure an audit device with a filter.
+
+<Note title="Advanced audit feature">
+  The use of filtering in Vault's audit system should be considered an advanced feature.
+  Whilst it has been designed to be simple and flexible, exclusively enabling filtered
+  devices could result in some requests and responses that are not audited.
+</Note>
+
+Please see the [Vault security model](/vault/docs/internals/security) for further information.
```
Comment (VioletHynes) [inline website/content/docs/concepts/filtering/audit.mdx:26]: I'm a little confused by this sentence -- we don't modify this file and it doesn't mention auditing in detail. This sentence feels like a non sequitur and I think we can probably remove it?

File: website/content/docs/concepts/filtering/audit.mdx [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+Filters can reference the following properties of an audit entry:
+
+* `mount_point` - path to the mount (e.g. [auth method](/vault/docs/auth), [secret engine](/vault/docs/secrets)),
+including namespaces.
+* `mount_type` - type of mount being interacted with.
+* `namespace` - [namespace](/vault/docs/enterprise/namespaces) 'path' that the request is taking place within.
+* `operation` - [operation](/vault/docs/glossary#operation) being performed.
+* `path` - full path of the request.
+
+### Example filters
```
Comment (VioletHynes) [inline website/content/docs/concepts/filtering/audit.mdx:47]: Love this section!

File: website/content/docs/concepts/filtering/audit.mdx [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+and [syslog](/vault/docs/audit/syslog)) support the `filter` option at the time they
+are enabled. After successfully enabling a device with a filter, every audit entry
+that Vault sends to that audit device will be compared to the expression in the filter.
+Only audit entries that match the filter should then be written to the device's audit log.
+
+### Valid `filter` properties
+
+Filters can reference the following properties of an audit entry:
+
+* `mount_point` - path to the mount (e.g. [auth method](/vault/docs/auth), [secret engine](/vault/docs/secrets)),
```
Comment (VioletHynes) [inline website/content/docs/concepts/filtering/audit.mdx:40]: This is a stylistic comment that I'm not 100% confident in, but I think we usually start bullet points off with a capital? I'm not sure which is 'right' though, so I'll definitely defer to the docs folks on this one, but I wanted to raise it at the very least

File: website/content/docs/glossary.mdx [hunk 8]
```diff
@@ -76,6 +77,33 @@ An external plugin may make use of [plugin multiplexing](/vault/docs/plugins/plu
...
+  or the Vault Namespace header.
+</Note>
+
+Please see our Enterprise documentation for further information on how
+[Namespaces can be specified](/vault/docs/enterprise/namespaces#vault-api-and-namespaces)
+as part of a request.
+
+#### Operation
+
+The requested operation, can be one of the following: `alias-lookahead`, `create`, `delete`,
```
Comment (VioletHynes) [inline website/content/docs/glossary.mdx:103]:
```suggestion
The request's operation can be one of the following: `alias-lookahead`, `create`, `delete`,
```

**Review by VioletHynes — COMMENTED** [review]
Looks great so far! I've left a few suggestions/comments but none of them are fundamental. I love the examples given and the structure of the new pages.

File: website/content/docs/concepts/events.mdx
Comment (peteski22) [reply website/content/docs/concepts/events.mdx:1]: No you're right, it was just the table formatting that was tweaked/sorted (and I think a numbering thing fixed too)

File: website/content/docs/concepts/filtering/audit.mdx (hunk 5 above)
Comment (peteski22) [reply website/content/docs/concepts/filtering/audit.mdx:26]: I wanted to include it in the `<note>` but it doesn't support the formatting. 😞 

The main reason being that the security model explains that all requests/responses to Vault will be audited (when audit is on), but through filtering it is possible for that not to happen, so I figured drawing attention to the page here might make sense 🤷🏼 🤔

**Review by VioletHynes — APPROVED** [review]
Looks great! :)

File: website/content/docs/concepts/filtering/audit.mdx (hunk 5 above)
Comment (VioletHynes) [reply website/content/docs/concepts/filtering/audit.mdx]: Hmm, yeah that makes sense. I'm definitely not opposed to leaving this in, but it does read a little strange right now. How about...

`To understand how auditing fits into the [Vault security model](/vault/docs/internals/security), please consult that documentation for further information.`

Though, it's fine as-is, just a suggestion :)

File: website/content/docs/concepts/filtering/audit.mdx [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+however the additional complexity can make it possible for a Vault Operator to configure their
+audit devices in such a way that some audit entries are missed out from audit logs entirely.
+
+The `fallback` audit device is the (non-mandatory) mechanism by which Vault can
+continue to adhere to Vault's [security model](/vault/docs/internals/security),
+that all requests and responses are successfully logged before the client receives
+any secret material.
+
+When exclusively using filtered audit devices, enabling the fallback audit device
+results in any audit entry which would have been ignored/missed due to filter settings
```
Comment (finnstech) [inline website/content/docs/concepts/filtering/audit.mdx:135]: This has a lot of passive voice so it's hard to parse. I suggest making it more active -- When exclusively using filtered audit devices, enable the fallback audit device to catch any events that would otherwise be missed.

**Review by finnstech — CHANGES_REQUESTED** [review]

File: website/content/docs/concepts/filtering/audit.mdx (hunk 5 above)
Comment (peteski22) [reply website/content/docs/concepts/filtering/audit.mdx]: I've updated it somewhere in between, thanks for the suggestions - they were really useful.

File: website/content/docs/audit/index.mdx [hunk 10]
```diff
@@ -140,6 +140,15 @@ docs](/vault/api-docs/system/audit) for more details.
...
   bodies](/vault/docs/audit#eliding-list-response-bodies) below.
 
+- `fallback` `(bool: false)` - If enabled, marks this audit device as the [fallback
+  device](/vault/docs/concepts/filtering/audit#fallback-device) in conjunction
+  with [filtering](/vault/docs/concepts/filtering/audit).
+  **Vault supports enabling only a single fallback audit device**.
+
+- `filter` `(string: "")` - Allows a predicate to be used by the audit device in
+  order to filter which audit entries this device should attempt to log.
+  See [filtering](/vault/docs/concepts/filtering/audit) for more information.
```
Comment (schavis) [inline website/content/docs/audit/index.mdx:150]:
```suggestion
- `filter` `(string: "")` - Sets an optional string used to filter the audit
  entries logged by the audit device.  See the
  [filtering](/vault/docs/concepts/filtering/audit) section of the auditing
  overview for more information.
```

Style correction: write in active voice, avoid "this" as a pronoun

File: website/content/docs/audit/index.mdx [hunk 11]
```diff
@@ -140,6 +140,15 @@ docs](/vault/api-docs/system/audit) for more details.
 - `elide_list_responses` `(bool: false)` - See [Eliding list response
   bodies](/vault/docs/audit#eliding-list-response-bodies) below.
 
+- `fallback` `(bool: false)` - If enabled, marks this audit device as the [fallback
+  device](/vault/docs/concepts/filtering/audit#fallback-device) in conjunction
+  with [filtering](/vault/docs/concepts/filtering/audit).
+  **Vault supports enabling only a single fallback audit device**.
```
Comment (schavis) [inline website/content/docs/audit/index.mdx:146]:
```suggestion
- `fallback` `(bool: false)` - Indicates whether the audit device is a
  fallback for filtering purposes. **Vault only supports one fallback audit
  device at a time**.
```

Style correction: write in active voice, avoid "this" as a pronoun

File: website/content/docs/commands/audit/enable.mdx [hunk 12]
```diff
@@ -40,3 +37,9 @@ flags](/vault/docs/commands) included on all commands.
 - `-path` `(string: "")` - Place where the audit device will be accessible. This
   must be unique across all audit devices. This defaults to the "type" of the
   audit device.
+
+There are a set of [common options](/vault/docs/audit#common-configuration-options)
+which can be applied to all types of audit device.
+
+For more details, please see the relevant page for an audit device `type`, under
+[Audit Devices docs](/vault/docs/audit).
```
Comment (schavis) [inline website/content/docs/commands/audit/enable.mdx:45]:
```suggestion
All audit devices support a
[common set of options](/vault/docs/audit#common-configuration-options), for
more details, refer to the specific audit device type in the
[Audit Devices](/vault/docs/audit) overview.
```

Style correction: write in active voice, simplify sentence structure when possible

File: website/content/docs/commands/events.mdx [hunk 13]
```diff
@@ -58,9 +58,8 @@ flags](/vault/docs/commands) included on all commands.
 - `-filter` `(string: "")` - Filter expression used to select events to be sent
   through the WebSocket.
 
-Refer to the [Filter expressions](/boundary/docs/concepts/filtering) guide
-in the Boundary documentation for a complete list of filtering options and an
-explanation on how we evaluate filter expressions.
+Refer to the [Filter expressions](/vault/docs/concepts/filtering) guide for a complete
+list of filtering options and an explanation on how we evaluate filter expressions.
```
Comment (schavis) [inline website/content/docs/commands/events.mdx:62]:
```suggestion
list of filtering options and an explanation on how Vault evaluates filter expressions.
```
Style correction: refer to the specific product/service/etc that takes action. Avoid generalizations like "we" unless speaking about things we (as people) do.

File: website/content/docs/concepts/filtering/audit.mdx [hunk 14]
```diff
@@ -0,0 +1,165 @@
+---
+layout: docs
+page_title: Audit Filtering
+description: |-
+  How to filter audit entries in Vault.
```
Comment (schavis) [inline website/content/docs/concepts/filtering/audit.mdx:5]: Is this actually a how to guide or a concept document? It's written as a concept doc (and under the concept section), but the description makes me think it's supposed to be a how-to? Let me know which you intended and I'll make edits appropriate to the doc type :)

File: website/content/api-docs/system/audit.mdx [hunk 15]
```diff
@@ -51,25 +51,28 @@ $ curl \
...
+- `description` `(string: "")` – Specifies a human-friendly description of the
   audit device.
 
-- `options` `(map<string|string>: nil)` – Specifies configuration options to pass to the audit device itself.
-  For more details, please see the relevant page for an audit device `type`, under [Audit Devices docs](/vault/docs/audit).
+- `options` `(map<string|string>: nil)` – Specifies configuration options to pass to the audit device itself.
+  There are a set of [common options](/vault/docs/audit#common-configuration-options)
+  which can be applied to all types of audit device.
+  For more details, please see the relevant page for an audit device `type`,
+  under [Audit Devices docs](/vault/docs/audit).
```
[the remark this answers is not in the document]
Comment (peteski22) [reply website/content/api-docs/system/audit.mdx:73]: Yes, each sub-page (file/socket/syslog) has options that are specific to it, this line was from the original documentation https://developer.hashicorp.com/vault/api-docs/system/audit#options

image

File: website/content/docs/concepts/filtering/audit.mdx (hunk 14 above)
Comment (peteski22) [reply website/content/docs/concepts/filtering/audit.mdx:5]: It's supposed to cover the concept of filtering specific to audit, with examples. I think that means the description isn't right. 😞

**Review by finnstech — APPROVED** [review]

File: website/content/docs/concepts/filtering/audit.mdx (hunk 14 above)
Comment (schavis) [reply website/content/docs/concepts/filtering/audit.mdx:5]: Okay, I'll edit it as a concept doc and suggest a new description for you :)

File: website/content/docs/concepts/filtering/index.mdx [hunk 16]
```diff
@@ -0,0 +1,128 @@
...
+### Values
+
+Values are used by matching operators to create an expression. Values can be any
+valid selector, a number, or a string. It is best practice to quote values.
+
+Numbers can be base 10 integers or floating point numbers.
+
+When quoting strings, they may either be enclosed in double quotes or backticks.
+When enclosed in backticks they are treated as raw strings and escape sequences
+such as `\n` will not be expanded.
```
Comment (schavis) [inline website/content/docs/concepts/filtering/index.mdx:72]:
```suggestion

Selector values can be any valid selector, integer, floating point number, or
string. Numbers and strings should be quoted in double quotes or backticks.

Strings quoted in backticks are treated as literal values and escape sequences
like `\n` are not expanded.

Value             | Type    | Expanded value
----------------- | ------- | ------------------------------ 
"Vault\tBoundary" | string  | "Vault	Boundary"
`Vault\tBoundary` | string  | "Vault\tBoundary"
"10"              | integer | "10"
`10`              | integer | "10"
"0.75"            | float   | "0.75"

```

File: website/content/docs/concepts/filtering/index.mdx [hunk 17]
```diff
@@ -0,0 +1,128 @@
...
+
+## Connect expressions
```
Comment (schavis) [inline website/content/docs/concepts/filtering/index.mdx:74]:
```suggestion
## Complex expressions
```

Suggest using "complex" expressions rather than "connecting" since there are compositions (like not) that may only include a single expression.

File: website/content/docs/concepts/filtering/index.mdx [hunk 18]
```diff
@@ -0,0 +1,128 @@
...
+
+There are several methods for connecting expressions, including
+
+- Logical `or`
+- Logical `and`
+- Logical `not`
+- Grouping with parenthesis
+- Matching expressions
```
Comment (schavis) [inline website/content/docs/concepts/filtering/index.mdx:82]:
```suggestion
Complex expressions combine basic expressions with logical operators, grouping, and matching expressions.
```

Suggest simplifying since you go into detail with the examples anyway

Comment (schavis) [reply website/content/docs/concepts/filtering/index.mdx:82]: Follow up question: what, exactly, is a "matching expression". The syntax example doesn't really make that clear. From context I'm guessing a matching expression is a regex? If so, are there limitations to the types of regex we support? Or a specific syntax we can reference?

File: website/content/docs/concepts/filtering/index.mdx [hunk 19]
```diff
@@ -0,0 +1,128 @@
...
+<Matching Expression 1>
+```
+
+Standard operator precedence can be expected for the various forms. For example,
+the following two expressions would be equivalent.
+
+```text
+<Expression 1> and not <Expression 2> or <Expression 3>
+
+( <Expression 1> and (not <Expression 2> )) or <Expression 3>
```
Comment (schavis) [inline website/content/docs/concepts/filtering/index.mdx:107]:
```suggestion
Vault uses standard operator precedence when  resolving complex
expressions. For example, the expression
`<Expression 1> and not <Expression 2> or <Expression 3>` resolves
the same as
`( <Expression 1> and (not <Expression 2> )) or <Expression 3>`.
```

Style correction: write in active voice

File: website/content/docs/concepts/filtering/index.mdx [hunk 20]
```diff
@@ -0,0 +1,128 @@
...
+```
+
+## Performance
+
+Filters are executed on Vault nodes and therefore will consume some amount
+of CPU time on the node.
```
Comment (schavis) [inline website/content/docs/concepts/filtering/index.mdx:113]:
```suggestion
Filters consume a portion  of CPU time on the Vault node where they run.
```

Style correction: write in active voice

File: website/content/docs/concepts/filtering/index.mdx [hunk 21]
```diff
@@ -0,0 +1,128 @@
...
+
+<Note title="Regular expressions">
```
Comment (schavis) [inline website/content/docs/concepts/filtering/index.mdx:115]: Performance question: comparing simple to complex is obvious, but what about a complex expression that joins multiple simple filters versus a single regex that captures the same thing?

File: website/content/docs/concepts/filtering/index.mdx [hunk 22]
```diff
@@ -0,0 +1,128 @@
...
+  Using multiple/complex expressions including regular expressions
+  (regex) will have a larger impact on performance than fewer/simpler filters.
+</Note>
+
+We recommend that all filters are tested in pre-production environments in order
+to ensure correctness.
```
Comment (schavis) [inline website/content/docs/concepts/filtering/index.mdx:121]:
```suggestion
Always test your filters in pre-production environments to ensure correctness.
```

Style correction: make best practices declarative.

File: website/content/docs/concepts/filtering/index.mdx [hunk 23]
```diff
@@ -0,0 +1,128 @@
...
+
+Ideally you should [codify your management of Vault](/vault/tutorials/operations/codify-mgmt-vault-terraform)
+using tools such as [Terraform](https://www.terraform.io/), to prevent accidentally enabling an audit device
+in a production environment with untested/incorrect settings.
```
Comment (schavis) [inline website/content/docs/concepts/filtering/index.mdx:125]: What does this have to do with filtering? o_O

File: website/content/docs/concepts/filtering/index.mdx [hunk 24]
```diff
@@ -0,0 +1,128 @@
...
+
+Finally, always ensure you profile production-like workloads within your pre-production
+environments in order to accurately assess the performance of Vault.
```
Comment (schavis) [inline website/content/docs/concepts/filtering/index.mdx:128]: Same question here. The last two points seem like general administrative/performance advice, not anything that's specific to filtering. I'm not saying it isn't potentially useful, just that this doesn't seem to be the right place for it.

File: website/content/docs/concepts/filtering/index.mdx [hunk 25]
```diff
@@ -0,0 +1,128 @@
+---
+layout: docs
+page_title: Filtering
+description: |-
+  An introduction to the filtering syntax used in Vault.
```
[the remark this answers is not in the document]
Comment (peteski22) [reply website/content/docs/concepts/filtering/index.mdx:5]: It also already applied to Events https://developer.hashicorp.com/vault/docs/concepts/events

File: website/content/docs/concepts/filtering/index.mdx (hunk 18 above)
Comment (peteski22) [reply website/content/docs/concepts/filtering/index.mdx:142]: Matching expressions are things like:

* `==`
* `!=`
* `is`
* `in`
* `contains`
* `matches` <-- this would be a regex

It comes from the HashiCorp bexpr library, which isn't well documented (there's a README but it doesn't break down all these expressions).

File: website/content/docs/concepts/filtering/index.mdx (hunk 23 above)
Comment (peteski22) [reply website/content/docs/concepts/filtering/index.mdx:181]: Using Terraform for the management of Vault config is a best practice (our Tao speaks to codification).

We felt it was important to have this here as a reminder/guide to users to help reduce potential issues.

Especially since the audit configuration is immutable (meaning turning it off and on can render prior HMAC values incomparable), and can possibly block Vault if configured incorrectly.

With this in mind, would it be OK to leave this in please?

File: website/content/docs/concepts/filtering/index.mdx (hunk 24 above)
Comment (peteski22) [reply website/content/docs/concepts/filtering/index.mdx:184]: This is because filtering is an advanced topic, and we want Operators to ensure they've profiled workloads on non-prod setups before going to prod. If they've done something like incorrectly configured lots of filters, this could lead to Vault trying to audit a lot and causing performance degradation. So we want to make sure we've given a friendly reminder that when you're in the world of filtering, you should be doing things by the book.

File: website/content/docs/concepts/filtering/index.mdx
Comment (peteski22) [inline website/content/docs/concepts/filtering/index.mdx:1]: this file is a port of the Boundary version (https://developer.hashicorp.com/boundary/docs/concepts/filtering)

File: website/content/docs/concepts/filtering/index.mdx (hunk 21 above)
Comment (peteski22) [reply website/content/docs/concepts/filtering/index.mdx:167]: We know that regexes are more complicated and so will take more time, but we don't have public data on the trade offs in mixing and matching etc. Below this we try to allude to customers being encouraged to test things out beforehand to get an idea for their workloads.

**Review by schavis — APPROVED** [review]
I'll make suggestions/edits to website/content/docs/concepts/filtering/audit.mdx under a separate PR.

**Review by finnstech — COMMENTED** [review]
LGTM
